### PR TITLE
Improve Memgraph's startup process

### PR DESCRIPTION
**Pull request by antoniofilipovic**
### Description

This diff applies changes from hypothesis testing from these two PRs:
- https://github.com/memgraph/memgraph/pull/2011
- https://github.com/memgraph/memgraph/pull/1999

to increase high-availability logging verbosity in e2e tests and wrap-around bolt server, websocket server and HTTP server ASSERTs so we are certain on which part has test failed.

Currently, there are a lot of silent fails, and no logging. The Bolt server should assert if it can't get started as an instance should have a dedicated port for the bolt server. Silent fails around HTTP and WebSocket servers are permissible.

`LoggerSink` won't be added to WSS if failure happened. If creating auth folder failed, Memgraph's initialization will be stopped.

[master < Task] PR
- [x] Provide the full content or a guide for the final git message
    - **Improve Memgraph's startup process**

### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - **If creating Bolt server fails, Memgraph won't be started. Failures in creating websocket and http server won't crash Memgraph. If creating websocket server fails, `LoggerSink` won't be added to it. Memgraph will crash if authentication folder cannot be created.**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [x] Tag someone from docs team in the comments

### CODE REVIEW

File: src/communication/v2/listener.hpp [hunk 1]
```diff
@@ -68,33 +76,63 @@ class Listener final : public std::enable_shared_from_this<Listener<TSession, TS
...
+           const uint64_t inactivity_timeout_sec)
+      : io_context_(io_context),
+        session_context_(session_context),
+        server_context_(server_context),
+        acceptor_(io_context_),
+        endpoint_{endpoint},
+        service_name_{service_name},
+        inactivity_timeout_{inactivity_timeout_sec} {
+    TryCreate<true>();
+  }
```
Comment (antoniofilipovic) [inline src/communication/v2/listener.hpp:93]: @Ignition I tried to make ctor be `template<bool AssertCreate>` but then I had some template issues and didn't input more time into this, let me know if you have any idea how to make this better

**Review by Ignition — APPROVED** [review]

**Comment by as51340** [comment]
@kgolubic RN is above, not sure if docs needed.

**Comment by antoniofilipovic** [comment]
@kgolubic RN: The Bolt server needs mapping to its port otherwise server won't start, metrics server and websocket server can have silent fail.

**Comment by kgolubic** [comment]
@antoniofilipovic I will use this as RN:

Memgraph will now fail to start if the Bolt server cannot be created, ensuring a dedicated port is available. Failures in creating the WebSocket and HTTP servers will no longer crash Memgraph but will be logged more effectively. If the WebSocket server fails, the `LoggerSink` will not be added to it. Additionally, Memgraph will crash if the authentication folder cannot be created.

**Pull request merged**